# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/visibility.rs
# Commit: aaa7e1f81404
# Full Hash: aaa7e1f8140455a9f83be8278720598ea3fe4cd8
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2021-12-14 09:42:05
# Description:
#   Bug 1745775 - Fix invalid surface info indexing r=gfx-reviewers,bradwerth
#   
#   Some parts of the visibility pass have been ported to use the
#   picture graph infrastructure (update pass assignment and bounding
#   rect propagation) but the main visibility pass still relies on
# ==============================================================================

diff -r de1608e5f69b -r aaa7e1f81404 gfx/wr/webrender/src/visibility.rs
--- a/gfx/wr/webrender/src/visibility.rs	Tue Dec 14 00:24:22 2021 +0000
+++ b/gfx/wr/webrender/src/visibility.rs	Tue Dec 14 01:06:24 2021 +0000
@@ -270,7 +270,7 @@
                 PrimitiveInstanceKind::Picture { pic_index, .. } => {
                     let (is_visible, is_passthrough) = {
                         let pic = &store.pictures[pic_index.0];
-                        (pic.is_visible(), pic.raster_config.is_none())
+                        (pic.is_visible(frame_context.spatial_tree), pic.raster_config.is_none())
                     };
 
                     if !is_visible {
